{
  "title": "When is EXCEPTION_REGISTRATION_RECORD created on the stack?",
  "link": "https://reverseengineering.stackexchange.com/questions/31559/when-is-exception-registration-record-created-on-the-stack",
  "content": "I want to understand when EXCEPTION_REGISTRATION_RECORDs get created on the stack.\n\nIs it true that there is one EXCEPTION_REGISTRATION_RECORD per try/catch?\n\n",
  "votes": "3",
  "answers": 1,
  "views": "99",
  "tags": [
    "seh"
  ],
  "user": "Bob5421",
  "time": "Feb 17, 2023 at 9:40",
  "comments": [
    {
      "user": "0xC0000022L",
      "text": "Could you please clarify one thing. You tagged it as seh, but you mention try/catch. Now in MSVC C++ exceptions are implemented in terms of SEH, but are you solely interested in the C++ side of things or in general?",
      "time": null
    },
    {
      "user": "Bob5421",
      "text": "I mean a MSVC C++ try/catch. I am interested in the wide general thing",
      "time": null
    },
    {
      "user": "peter ferrie",
      "text": "I replaced my original answer because I was somehow describing the exception_record rather than the exception_registration_record.",
      "time": null
    }
  ],
  "answers_data": [
    {
      "content": "EXCEPTION_REGISTRATION_RECORD is placed on the stack in anticipation of an exception occurring.  The \"try\" will put one there.  The record is part of a chain, for as many nested \"try\" statements as exist.  The nesting includes one function calling another function from within a try block.  Windows will also put a top-most handler there before the program starts, so there is always one on program start.\n\n",
      "votes": "4",
      "user": "peter ferrie",
      "time": "Feb 25, 2023 at 23:24",
      "is_accepted": true,
      "comments": []
    }
  ]
}